src/util/simplify_expr.cpp

```diff
@@ -236,22 +236,34 @@ bool simplify_exprt::simplify_typecast(exprt &expr)
     return false;
   }
 
-  // eliminate casts to _Bool
-  if(expr_type.id()==ID_c_bool &&
-     op_type.id()!=ID_bool)
+  // circular casts through types shorter than `int`
+  const auto check_circular_cast=[&](const typet &type)
```

Comment:
Given the length and non-triviality of this lambda I'd suggest pulling it into its own function.

Reply:
Turns out it can be written as a for-loop, which means we don't need to pass a bunch of local variables into a whole new function.

Reply:
Where's the for loop?